### PR TITLE
Implement /profile handler (GET only).

### PR DESCRIPTION
**Pull request by Half-Shot**
A readonly handler for now for /profile, according to the spec.
Including a misc change for main.py

### CODE REVIEW

File: bluepill/server.py [hunk 1]
```diff
@@ -50,8 +71,36 @@ def stop(self):
         ioloop.add_callback(self.ioloop.stop)
         self._thread.join()
 
+
 # TODO: Actually organise and name the methods properly.
 class MatrixClientVersions(RequestHandler):
     def get(self):
         response = {'versions': ['r0.1.0']}
         self.write(response)
+        # TODO: Actually organise and name the methods properly.
```
Comment (SShrike) [inline bluepill/server.py]: Why is this comment here? Is it a leftover from a copy-paste?

File: bluepill/server.py [hunk 2]
```diff
@@ -50,8 +71,36 @@ def stop(self):
...
+                    }
+                )
+        elif match('@.+:.+', user) is not None:
+            self.set_status(404)
+            self.write(getErrorMessage('M_NOT_FOUND', "Could not find user."))
+        else:
+            self.set_status(400)
+            self.write(getErrorMessage('M_UNKNOWN', "Bad Userid"))
+
+    def post(self,user,request):
```
Comment (SShrike) [inline bluepill/server.py]: Missing spaces between parameters.

File: bluepill/server.py (hunk 1 above)
Comment (Half-Shot) [inline bluepill/server.py]: That's yours? Might have been the result of a automerge -_-

File: bluepill/server.py [hunk 3]
```diff
@@ -50,8 +71,36 @@ def stop(self):
...
+            else:
+                self.write(
+                    {
+                        'displayname': dummy['displayname'],
+                        'avatar_url': dummy['avatar_url']
+                    }
+                )
+        elif match('@.+:.+', user) is not None:
+            self.set_status(404)
+            self.write(getErrorMessage('M_NOT_FOUND', "Could not find user."))
```
Comment (SShrike) [inline bluepill/server.py]: Single quotes and double quotes are mixed, stick to single quotes.

File: bluepill/server.py [hunk 4]
```diff
@@ -50,8 +71,36 @@ def stop(self):
...
+        else:
+            self.set_status(400)
+            self.write(getErrorMessage('M_UNKNOWN', "Bad Userid"))
```
Comment (SShrike) [inline bluepill/server.py]: Same as above.

File: bluepill/server.py [hunk 5]
```diff
@@ -14,16 +14,37 @@
 # limitations under the License.
 
 import threading
-
+from re import match
 from tornado.ioloop import IOLoop
 from tornado.web import Application, RequestHandler
 
+_dummyUser = {
+    'userId': "@alice:example.com",
```
Comment (SShrike) [inline bluepill/server.py]: Single quotes and double quotes are mixed, stick to single quotes.

File: bluepill/server.py [hunk 6]
```diff
@@ -50,8 +71,36 @@ def stop(self):
...
         response = {'versions': ['r0.1.0']}
         self.write(response)
+        # TODO: Actually organise and name the methods properly.
+
+
+class MatrixClientProfile(RequestHandler):
+    def get(self, user, request):
+        dummy = getDummyUser()
+        if dummy['userId'] == user:
+            if request == "displayname":
```
Comment (SShrike) [inline bluepill/server.py]: Use single quotes instead of double quotes.

File: bluepill/server.py [hunk 7]
```diff
@@ -50,8 +71,36 @@ def stop(self):
...
+                self.write({'displayname': dummy['displayname']})
+            elif request == "avatar_url":
```
Comment (SShrike) [inline bluepill/server.py]: Same as above.

**Comment by SShrike** [comment]
LGTM for the most part. My final nit is that you need to remove the double spaces between classes and stuff, I have the PEP8/flake8 warning disabled in the `setup.cfg`.

**Comment by Half-Shot** [comment]
> LGTM for the most part. My final nit is that you need to remove the double spaces between classes and stuff, I have the PEP8/flake8 warning disabled in the setup.cfg.

Okay, figured that might have been the case. I'll adjust my pep locally.

> Also, rebase changes from now on, so that we avoid a whole bunch of commits like ''Merge remote-tracking branch 'origin/master' into HS-Develop".

Sure.

**Comment by Half-Shot** [comment]
Why would you not have a merge and close button next to the comment button :(

**Pull request merged**